File: frontend-v1/types/record.ts

// =======================
// Record Request DTO
// =======================
export interface RecordRequest {
  patientId: number;          // int64 -> number
  appointmentId: number;      // int64 -> number
  visitDate: string;          // ISO date string (YYYY-MM-DD)
  diagnosis: string;          // 0 - 500 characters
  treatmentPlan: string;      // 0 - 1000 characters
  notes: string;              // 0 - 2000 characters
}

// =======================
// Record Response DTO
// =======================
export interface RecordResponse {
  recordId: number;           // int64 -> number
  patientId: number;
  patientName: string;
  appointmentId: number;
  visitDate: string;          // ISO date string
  diagnosis: string;
  treatmentPlan: string;
  notes: string;
}

File: backend/src/main/java/com/backend/model/Record.java
package com.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "records")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Record {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "record_id")
    private Long recordId;

    // Many records belong to one patient
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "patient_id", nullable = false)
    private Patient patient;

    // Optional relationship to appointment
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "appointment_id")
    private Appointment appointment;

    @Column(name = "visit_date", nullable = false)
    private LocalDate visitDate;

    @Column(name = "diagnosis", nullable = false, length = 500)
    private String diagnosis;

    @Column(name = "treatment_plan", length = 1000)
    private String treatmentPlan;

    @Column(name = "notes", length = 2000)
    private String notes;
}
